backend-diagram-converter/core/src/main/java/org/camunda/community/migration/converter/visitor/impl/attribute/CollectionVisitor.java

```diff
@@ -29,7 +29,20 @@ protected Message visitSupportedAttribute(DomElementVisitorContext context, Stri
                 .getZeebeLoopCharacteristics()
                 .setInputCollection(transformationResult.getNewExpression()));
     context.addMessage(MessageFactory.collectionHint());
-    return MessageFactory.collection(
-        attributeLocalName(), context.getElement().getLocalName(), transformationResult);
+    Message message;
+    if (transformationResult.hasExecution()) {
+      message =
+          MessageFactory.collectionExecution(
+              attributeLocalName(), context.getElement().getLocalName(), transformationResult);
+    } else if (transformationResult.hasMethodInvocation()) {
```

Comment:
This should be an if-logic without else as the cases do not exclude each other. This would lead to potentially multiple messages added per expression, but the result is deterministic.

Reply:
See my comment above.